### ISSUE TITLE
Split MNX-Common and MNX-Generic into two separate formats

### ISSUE THREAD

**Issue by adrianholovaty** [w3c/mnx#98]
I've been concerned about how confusing it would be for end users to have to know the difference between MNX-Common and MNX-Generic.

Faced with a "Export..." dialog box that has two separate MNX formats, the majority of non-technical users wouldn't have the slightest idea why to choose MNX-Common vs. MNX-Generic. Heck, I'm a programmer of music-notation software who has been following the specification's progress, and _even I_ would have a hard time explaining the difference to a layperson.

Some anecdotal evidence: For my own product, which ingests MusicXML, I've gotten anxious emails from customers asking "Should I export as compressed or uncompressed MusicXML?!" Fortunately, the answer there is "We support both" — because the underlying notation data is identical and the only difference is that one of the files is zipped — but this won't be the case for the two MNX formats, which are radically different in meaning and scope.

I chatted about this with @joeberkovitz a few days ago at Musikmesse, and he explained his assumption was that there would be a single container format, so end users would only need to "export as MNX." That makes sense at face value, and I suggest documenting that more obviously. But I think this may merely be kicking the can down the road.

I think it would be a great benefit to the project to have clear answers to the following questions:

- Are MNX-producing applications expected to export MNX files that contain both MNX-Common and MNX-Generic?
    - If yes, then that's likely a nonstarter... Apps that don't deal with graphical display will have no way to generate MNX-Generic, and apps that deal without semantics won't have a way to generate MNX-Common.
    - If no, then how does an MNX-producing application communicate the following to end users:
        - Which type(s) of MNX it's capable of producing.
        - The difference between the MNX flavors (so that users don't just blindly choose one).
- Given an MNX file, how will an end user be able to quickly determine whether it's MNX-Common, MNX-Generic or both?
- How does an MNX-consuming application communicate which flavor(s) of MNX it supports?

The more I consider this, the more I think MNX could be a colossal failure in the marketplace if we don't address these issues clearly and cohesively. :-/ I have come to believe that using a single user-facing "brand" to handle both MNX flavors would be a bad idea. The only way to avoid mass confusion would be to have clearly distinct concepts — distinct file extensions, distinct "Export as..." options, distinct "Import" options.

It could very well be that this is the plan (I hope it is!); if so, let's make that crystal clear in the documentation.

**Comment by clnoel** [comment]
My impression was that MNX-Generuc, MNX-Common, and the other TBD flavors are considered separate products, which is why we can focus on just one at a time. So an export program or a cosumer can't say that it supports MNX... There's no such thing. For the rest, so far I have only seen internal file tags to determine what is actually cointained in the xml. Since they are just xml files, they could have any extension, I guess. I'm okay with specifying ecpected extensions, but it's got to be the inside that determines compatibility.

This is complicated by the MNX-Common profiles that we are developing. Not all MNX-Common files will be compatible with all readers. But if we do those profiles right, the programs should be able to state what profiles they support as well. After all, you know whether your program can handle multi-metric, TAB notation, etc. The profiles will be along those lines.

**Comment by joeberkovitz** [comment]
@adrianholovaty Very good issue to raise, and my suggestion of a single "Save as MNX" (along with various saving options) was off-the-cuff, although it does reflect my rough thinking so far. A single unifying label might be a bad idea, and I do see that that having different and clearly distinguished file formats is definitely one viable way to go. But let's try to chew on the unified concept a little more before tossing it. I promise not to hold tight to the idea if it's truly going nowhere :-)

First of all, let's agree that *requiring* export of MNX-Generic is indeed a bad idea. Moving on...

The options for exporting _from a notation editor_ likely boil down to this:

- Save MNX-Common only
- Save MNX-Common along with MNX-Generic graphics (that link back to the Common piece)
- Save MNX-Common along with MNX-Generic performance only (also with linkage)
- Save MNX-Common along with MNX-Generic graphics _and_ performance (also with linkage)

It's starting to look as the the UI for a "Save As MNX" button might therefore appear as two checkboxes, maybe three if you count performance data:

[  ] include graphical score
[  ] include playback audio

Not too bad -- seems readily intelligible. And let's assume that these various options all wind up being packaged as a single file, not an unwieldy folder full of things. However, then we still have the problem of making end users able to understand just what flavors of MNX a consuming app is able to consume (so that they know which boxes to check in a UI like the above). If everything is just called "MNX", that could be challenging, as you say.

Would love to hear more comments on this point.

**Comment by joeberkovitz** [comment]
@clnoel By the way, I think having a ton of different profiles has all of the above problems and even more, because we can't name a format "MNX-Common-Single-Metric-Implied-Accidentals-And-Chocolate-Frosting" and expect people to keep up :-) (I know you didn't propose this, just taking it to an extreme.)

The chairs as a whole are not sold on  letting each app decide what profile(s) to use, in a kind of free-for-all. That way lies probable incompatibility and failure. The proposed profile mechanism is more of a way to say, "This app breaks compatibility with the standard profile, and here's exactly how: ...".

**Comment by clnoel** [comment]
Well, yes. That's the point of having a "well-behaved" profile in the first place. The problem I see is that there is a huge corpus of music that is not "well-behaved" in that it is multi-metric, or that it has tablature, which I don't think a base MNX program needs to be able to read. I think that anything saying it can export/import MNX-Common needs to be able needs to be able to deal with that well-behaved profile, whatever we decide it is. The issue we've got here is how do two programs who both handle multi-metric let each other know that it's okay to break that profile constraint?

**Comment by notator** [comment]
It ought to be possible to find some middle ground if we distinguish clearly between profiles and file formats.
Profiles could be organized in a hierarchic tree structure [1], but only a few of them need to correspond to public file formats.
Note that applications _always_ decide which file formats they are going to support, and that users select the applications that match their needs (=use-cases).

There could, for example, be four file formats, each of which is related to a particular use-case, has its own file extension, and is therefore recognisable from the outside:
1. .mnx1: MNX-Generic -- links an amorphous graphic with temporal information (see [2] , [3] and [4])
2. .mnx2: MNX-Common -- the standard file format (here including multi-metric etc., but not TAB).
3. .mnx3: MNX-Common -- some other CWMN file format that is incompatible with .mnx2 (see [5])
4. .mnx4: MNX-TAB -- tablatures (see [6])

A user who wants to edit .mnx1 files probably just wants to synchronize a scan of an old manuscript with a demonstration recording, so doesn't need an editor that is designed for .mnx2. Similarly for .mnx2, .mnx3 and .mnx4. See "hacks" in the last part of https://github.com/w3c/mnx/issues/79#issuecomment-373715487.

To become a standard, each of these file formats has to be implemented by at least one (or two?) reference application(s). For example, Dorico could be a reference application for .mnx2. (Dorico would not have to implement either .mnx1, mnx3 or .mnx4.) In effect, Dorico would _define_ what MNX-Common's standard profile should contain. If Dorico decided not to support multi-metric scores, then multi-metric scores would have to be part of a different file format.

It would still be possible for a simpler editing application (e.g. an application specialised for church choir music) to import/export the .mnx2 format. If such an application sees information that it can't or doesn't want to handle (e.g. nested tuplets), then it should just ignore it gracefully (i.e. reject the file or create some kind of graphic equivalence).

Similarly, if Dorico was a reference application, and it decided that multi-metric was part of the .mnx2 file format, its implementation could (initially) be to ignore the information gracefully.

The bottom line seems to be that file formats should be related to well-defined use cases, and that we need to define the profile hierarchy (probably a tree , in #68, #55, #69 etc.) before deciding which profiles should be published as file formats.
But maybe this could be done partly the other way round: First simply _decide_ what the standard file format (.mnx2) should support, then extract the profile tree in order to find out which other file formats need to be defined.

---
**Footnotes**
(Some of these should be raised as issues. @joeberkovitz: Which of these should I raise as new issues, and which can be appended to existing ones?)
[1] Proposal: The root profile in the CWMN tree should contain just the characteristic chord symbols, rests, staves and clefs. That's not enough to define a file-format.
[2] Proposal: An identically defined <temporal-information> element should be included at the top level in all the MNX file formats (see #95) . MNX-Generic should be no exception to this rule.
[3] I still don't understand why one would want to include both <mnx-generic> and an <mnx-common> profile in the same file. Does [2] affect this?
[4] I'm not happy with the name MNX-Generic. The Group Charter still calls this format "GMNX", so its not too late to revise this decision. I'd prefer the name MNX-Graphic.
[5] .mnx3 could, for example, omit tuplets and metronome marks, and therefore be simpler than .mnx2. (see #79).
[6] .mnx4 could probably cope with non-Western tablatures as well, and so go a long way to internationalizing this project.

**Comment by joeberkovitz** [comment]
The chairs need to discuss this and come back with some clear guidance.

I can say this much: in the chairs' previous discussions we have been extremely wary of allowing a large number of profiles, which would likely lead straight back to the same application-specific support difficulties we see today with MusicXML. SVG has something like 4 profiles, and even these confuse people attempting to export SVG from various tools.

**Comment by shoogle** [comment]
I don't recommend going down the container route and using the `.mnx` extension for an archive that can contain any kind of document. This is bad enough for media containers:

- __User:__ Why can't I open this `.mp4` file?
- __Developer:__ We don't support the Vorbis audio codec.
- __User:__ But the application says it works with `.mp4` files?!?!

At least for media containers there is the *possibility* that the user could install the required codec and then it might work (the file is still just audio and/or video data after all). However, MNX-Common and MNX-Generic are two entirely different things that serve different purposes. There are legitimate reasons why an application might be able to import one and not the other, I think they should use different file extensions (e.g. `.mnc` and `.mng`).

However, I wonder if the problem extends beyond file extensions. The names "MNX-Common" and "MNX-Generic" are too long, so people are inevitably going to drop the endings and refer to them both as "MNX". It might be worth refering to them as "MNC" and "MNG" ourselves to prevent the distiction being lost.

**Comment by jsawruk** [comment]
After reading the co-chair meeting minutes from March 5, 2019, I agree with @adrianholovaty and @mdgood  that having to support multiple formats may be confusing for all of the reasons he listed. I currently have the opinion that the group should focus on MNX-Common and create a separate format for MNX-Generic that does **not** use the name MNX (to avoid confusion). They could potentially share a common metadata format (MNX-Metadata?), but otherwise I think they should be as distinct and distinguishable as possible.

**Comment by notator** [comment]
I've also read the co-chair meeting minutes from March 5, 2019, and agree very much with the co-chair and @jsawruk's comments above.
More precisely: I think that the MNX container idea should simply be dropped, and that it should be made very clear that MNX-Common and MNX-Generic are two completely different formats.

The minutes say that
- @adrianholovaty  "thinks it’s crucial to do the job of MNX-Generic but the issue is how the two formats are positioned."
- "The co-chairs are in broad agreement that we should focus MNX on the semantic format currently known as MNX-Common and move the MNX-Generic format to a separate specification."

Most active CG members have a MusicXML background, so concentrating their energies on MNX-Common would obviously be a good idea.
I think there is a window of opportunity for creating MNX-Generic*, but that that window will close if I'm the only one left wanting to discuss the subject. We either have to admit that MNX-Generic is going to fail, or get more people involved in discussing it.


*I currently think of MNX-Generic as being SVG containing specialised (namespaced) temporal information and object classes.

**Comment by dspreadbury** [comment]
@notator It's important to note that some aspects of MNX-Generic are already specified, so it doesn't matter how you currently think of it – it already exists in broad strokes.

**Comment by adrianholovaty** [comment]
Here's an email I've just sent to the community group's mailing list:

-----

Hi all,

In defining the MNX format, we've envisioned having a single format that contains at least two "types" of music notation data — MNX-Generic and MNX-Common. I've come to believe this isn't the right move.

(For background, see the [current spec](https://w3c.github.io/mnx/specification/).)

I believe MNX-Generic and MNX-Common are sufficiently different that they should just be two separate formats, rather than us trying to shoehorn them into a single format. Three reasons:

1. It's confusing for end users. Which one should users choose when importing/exporting? How would the user understand the differences in the level of semantics between the two similarly named formats? In my own experience, it's already confusing enough explaining the difference between a PDF and MusicXML, and those have different names!

2. It's confusing for developers. The type of development work required to parse something that is essentially an SVG wrapper versus a semantic document model is completely different. I would estimate 90% of developers would only want/need to parse one or the other — which strongly suggests they should be separate formats.

3. There's no obvious benefit in combining them, apart from some negligible things like having a common metadata format.

With this in mind, I'm proposing we formally split this into two formats, with distinct names and distinct specs. (To be clear, I think there's clear value in both formats existing.)

I brought this up with the other co-chairs in our latest meeting and we're in broad agreement. [Here are the meeting minutes](https://www.w3.org/community/music-notation/2019/03/05/co-chair-meeting-minutes-march-5-2019/) with more details.

Can anybody articulate a reason why we shouldn't make this change? I'd like to make sure we aren't overlooking a benefit that a single format would give us. If not, I can begin the work of splitting the spec in two.

Cheers,
Adrian

**Comment by JDLH** [comment]
My recollection (and I haven't gone back to the list archives to confirm this) is that the main thrust of MNX and this community group is Common Western Musical Notation of a scope that satisfies an 80/20 rule: the small fraction of the complexity which satisfies the large majority of the use cases. However, there was a determined opinion on the list that MNX should also offer something for the 20/80 case: the minority of the use cases, especially including contemorary and unconventional Western Music Notation, which is much more complex. 
The fallback to SVG graphical content was, I interpreted, giving up on the idea of describing the symbolic content of such notation, and aiming only to preserve a visual appears to humans for some presentation layout context (a page which is x cm wide and y cm tall, for example).
My bias is, that the use cases I really care about are solidly based on representing the symbolic content of the notation well. I want to allow a) analysis and reuse of notation content by software, most notably a "diff" tool that finds similarities and differences in musical content; and b) musician's apps for displaying, rehearsing, and performing music scores, which can revise layout in response to user interventions — most notably hiding an entire set of instrumental parts, or revising the layout in response to the user turning the tablet screen from portrait to landscape orientation. I can't imagine that an SVG container would support those use cases very well, but I'm willing to be surprised.

**Comment by bhamblok** [comment]
I would love to see though to end up with all of this in such a way we can mix up both formats...

Imagine a piece of work which combines CWMN and eg: graphical representations of audio-envelops including controller-data for an electronic instrument used in live performances.
I see great benefits if we could use small SVG's (enriched with musical semantic data) for some parts in a score, for some measures in a score, on top of CWMN in other parts at the same moments (or in next measures for the same part...).

I think that might have been an argument to combine the two formats into one spec.
(Did I explain myself in a clear way?)

We should at least be able to make a reference to another file in our CWMN-encoded MNX-Common-file, at a specific time for some part, wrapped in some kind of "container"-tag having some semantic attributes. Or is this completely out of scope of MNX CWMN?

I do think we have the opportunity here, to prepare ourselves for "the future" of music notation.

**Comment by shoogle** [comment]
Funny thing is both PDF and SVG support embedded audio, animations and interactivity, so perhaps we don't need another generic format?

**Comment by jsawruk** [comment]
@shoogle: Those are my thoughts exactly. There are already container formats like PDF, and you can create interactive documents using HTML. I'm not sure we need yet another container format when those two alone can provide a lot of use cases.

I think it really comes down to musical semantics. MNX-Common is a format that captures musical semantics such as pitch and rhythm, but I don't fully understand how that would work in MNX-Generic. 

Would it be possible to allow SVG documents within MNX-Common, in the same way that SVG documents can be embedded in HTML documents? If we wanted to add musical semantics to the embedded SVG, might it be possible to create a separate namespace for those semantics?

**Comment by mscuthbert** [comment]
Per Jeremey's request to consolidate discussion.  On email I wrote:

> One principal reason for keeping both frameworks and to keep them under a single name is for transition from MusicXML to MNX for researchers who would otherwise be considering MEI.  How will we indicate that MNX (or whatever the overarching name is called) is a format that can handle representation of the 2nd and 3rd scores in https://joeberkovitz.github.io/gmnx-viewer/  ?   I think that there are important marketing considerings for MNX for the research community to be considered with respect to “MEI can do X, can MNX do it?” which should be answered with “yes.” and not “no, MNX can’t, but Y can, and it’s by the same people”.  What will the other future expansions of MNX previously discussed (MNX-Neumes, etc.) be called?  I need to know a little more, but for now, I’m -1 on the idea.

**Comment by notator** [comment]
@dspreadbury In the past, there's been a lot of confusion about what "MNX-Generic" means. I just wanted to make it clear that I'm not interested in discussing a version that is "relatively free of semantics".

@adrianholovaty I think the definition of MNX-Generic (at the top of the [current spec](https://w3c.github.io/mnx/specification/)) will have to be changed, but that that can wait until after you have finished your first revision of the spec.

@jsawruk It is possible to add arbitrary semantics (class names for arbitrary graphical objects and temporal information) to SVG files, so it should be possible to preserve all the semantic information in an MNX-Common document in an MNX-Generic (semantically rich SVG) document containing the same score. The difference between the two formats would simply be that MNX-Generic contains an _instantiation_, while MNX-Common just contains the _logical_ _structure_. I think MNX-Common might eventually become redundant, but that it would be a good idea to develop the two formats in parallel for the moment to ease the transition from MusicXML, -- and in case I'm wrong! :-)

@JDLH MNX-Generic can also contain meta-information as to the type of notation being encoded. So it can also cope efficiently with non-CWMN notations, sharing class names between notation types where possible. That increases the scope and user-base considerably. Asian notations, for example... Be prepared to be surprised! :-)

@bhamblok I think MNX-Common should contain a purely logical description of the score. Making a reference to an external file from MNX-Common would be mixing up the logic with the instantiation.
For example: MNX-Common might say that there is a slider with a series of values that applies to a series of chords, but it should be left to the SVG-instantiating application to decide how the slider actually looks _in space_. Similarly, its up to the performing application to decide what the slider's values actually mean _in time_.

@shoogle I think its important that the symbols in music scores don't move about while we are looking at them. So we can ignore SVG's animation capabilities -- at least for now. :-)

@mscuthbert _Any_ score can be represented in SVG, including those containing neumes and those at [https://joeberkovitz.github.io/gmnx-viewer/](https://joeberkovitz.github.io/gmnx-viewer/). Its just a question of finding an appropriate score editor/instantiator. There are SVG versions of scores 1 and 3 working at [Fauré: Après un rêve](http://james-ingram.de/altMNX/cursors/demos/verticalLineCursorDemo/verticalLineCursorDemo.html) and [Lutoslawski](http://james-ingram.de/altMNX/cursors/demos/multiRectCursorDemo/multiRectCursorDemo.html).

**Comment by shoogle** [comment]
I think the most important point is to avoid confusion for users and application developers. We don't want a situation where application developers can completely ignore MNX-Common, export CWMN scores as MNX-Generic, and claim that their application supports MNX. This example should be enough to see that the formats cannot be combined.

@mscuthbert
> One principal reason for keeping both frameworks and to keep them under a single name is for transition from MusicXML to MNX for researchers who would otherwise be considering MEI. How will we indicate that MNX (or whatever the overarching name is called) is a format that can handle representation of the 2nd and 3rd scores in https://joeberkovitz.github.io/gmnx-viewer/ ?

Can you elaborate on how MEI would represent these scores? Does it have the ability to represent these scores semantically, or does it fallback to using a non-semantic representation of "here's what it looks like" and "here's what it sounds like" for non CWMN scores, as proposed in MNX-Generic?

> “MEI can do X, can MNX do it?” which should be answered with “yes.” and not “no, MNX can’t, but Y can, and it’s by the same people”

If the MEI representation of X is semantic then the non-semantic representation proposed by MNX-Generic will be no substitute for it. If the MEI representation is non-semantic then I would argue that MEI cannot do X properly, so it would be wrong for us to follow their bad example and claim that we support something when in fact we do not. As I said before, doing so risks confusing users and allowing application developers to ignore the semantic features entirely.

I would prefer to see MNX-Common define a standard set of semantic tags that applications must support to meet the 80/20 rule proposed by @JDLH. However, rather than relegating the remaining 20% of cases to a non-semantic format, I would prefer these were dealt with as optional extensions (or "profiles") to MNX-Common. The extensions mechanism would be defined in the core MNX-Common specification, so applications could detect when a score contains features they don't support and display a meaningful error message to the user.

@notator 
>  I think its important that the symbols in music scores don't move about while we are looking at them. So we can ignore SVG's animation capabilities -- at least for now. :-)

The point of a generic format is that it doesn't impose any restrictions on notation. For example, you might want to display a moving pianoroll that is synced with the audio. However, animation is still useful even for ordinary notation, as it allows you to display a playback cursor. See the examples at https://joeberkovitz.github.io/gmnx-viewer/

**Comment by jsawruk** [comment]
@shoogle:

> We don't want a situation where application developers can completely ignore MNX-Common, export CWMN scores as MNX-Generic, and claim that their application supports MNX.

This is a very strong point. I completely agree.

> The extensions mechanism would be defined in the core MNX-Common specification, so applications could detect when a score contains features they don't support and display a meaningful error message to the user.

I like this idea too, it's basically what MIDI 2.0 does with MIDI-CI: "MIDI-CI separates older MIDI products from newer products with new capabilities and provides a mechanism for two MIDI devices to understand what new capabilities are supported." Basically, it allows for baseline MIDI 1.0 functionality and queries about additional profiles that the device may support. Similarly, having a baseline set of CWMN semantics with additional add-on profiles (maybe by using namespaces?) might be a way to extend that 80/20 rule. It won't cover all use cases, but might get closer to 85/15 or 90/10. The problem with this approach is that the additional features would need to be specified as well, but I do like that there could be some optional components of MNX-Common.

**Comment by shoogle** [comment]
> The problem with this approach is that the additional features would need to be specified as well

That's true, but that can wait until the main specification is finalised. The aim would be to make the core set of features something that applications can realistically support in its entirety, rather than saying "we support MNX-Common" when actually they only support a subset. Applications would then opt-in to the particular extensions that they feel they are willing and able to support properly, and that are relevant to their users.

**Comment by notator** [comment]
@shoogle Interesting, stimulating points. Thanks.
 
> We don't want a situation where application developers can completely ignore MNX-Common, export CWMN scores as MNX-Generic, and claim that their application supports MNX.

I agree with you completely if you mean the non-semantic MNX-Generic as defined in the current spec. That's a very good reason for not continuing to work on non-semantic MNX-Generic.
On the other hand, there's no problem if MNX-Generic and MNX-Common share the same logical structure and semantic information. Developers would then not be able to ignore MNX-Common and export MNX-Generic.

>The extensions mechanism would be defined in the core MNX-Common specification, so applications could detect when a score contains features they don't support and display a meaningful error message to the user.

The extension mechanism could be copied almost verbatim into an MNX-Generic (semantically enhanced SVG) file. But note that the notation types defined for MNX-Common would not be the only types defined for MNX-Generic. If you are instantiating a CWMN notation type in MNX-Generic you have to declare which type it is, but you don't at the same time have to know anything about the notation types that you are not writing. That means that the types supported by MNX-Generic can be extended without changing the MNX-Common definition -- which ought to please both MNX-Common programmers and programmers of any new MNX-Generic types. Seamless extensibility is the name of the game! :-)

> The aim would be to make the core set of features something that applications can realistically support in its entirety, rather than saying "we support MNX-Common" when actually they only support a subset. 

Yes, exactly. I think MNX-Common would be more stable, and much easier to finalise, if it defined a fairly small set of notation types.

>The point of a generic format is that it doesn't impose any restrictions on notation. For example, you might want to display a moving piano roll that is synced with the audio. However, animation is still useful even for ordinary notation, as it allows you to display a playback cursor.

There are easier ways to scroll SVGs, or display running cursors, than to use SVG's animation capabilities. Note that the graphical objects in a scrolling piano roll do not move with respect to each other, and that cursors (lines, rectangles etc.) can be moved simply by changing their physical coordinates relative to the score. My [Fauré](http://james-ingram.de/altMNX/cursors/demos/verticalLineCursorDemo/verticalLineCursorDemo.html) and [Lutoslawski](http://james-ingram.de/altMNX/cursors/demos/multiRectCursorDemo/multiRectCursorDemo.html) examples don't use SVG animation.
Of course, there's nothing to stop a performing application using SVG animation to implement smoothly moving cursors if it wants to. I just don't think that's something that needs to go in the score definition. Maybe I'm wrong, but I think there are good, _fundamental_ reasons why symbols in scores don't move about with respect to each other! :-)

An afterthought to my previous posting:
I think MNX-Common should be developed with an eye to how applications are going to use the information to implement score instantiations. There's no point in developing a format that is going to be too difficult to use. So I think its proposed features should always be tested (in plausible thought experiments, if not concretely) while in development. That means I think its a bad idea to try to forget about MNX-Generic (semantically enhanced SVG) while developing MNX-Common. They can't really be thought of as two really separate formats as per this issue's subject line. Non-semantic MNX-Generic, as defined in the current spec should, like the MNX container, be dropped completely.

**Comment by mdgood** [comment]
@notator I thought we had been as clear as we could be in the group charter and other previous communications, but apparently not. So let me restate this again. 

We are not working on a semantically enhanced SVG format in this community group. That is not what MNX-Common is. You are welcome to develop such a format yourself, and I understand its appeal for your own work. But it is outside the scope of this group and this GitHub issue. As stated in the charter, MNX-Common is designed for applications that do not depend on particular notational systems or their semantics.

**Comment by notator** [comment]
@mgood I have never said that MNX-Common is semantically enhanced SVG.

> As stated in the charter, MNX-Common is designed for applications that do not depend on particular notational systems or their semantics.

@mgood I think you meant MNX-Generic there.

The charter says that the MNX specification consists of three parts:
>The MNX container format, which can describe and relate an arbitrary number of components and documents which collectively describe a notated work as a whole.

Do we all, including the co-chair, agree that the MNX container format is redundant?

>The CWMNX format (working title), designed to represent Common Western Music Notation in a tightly specified, semantic fashion in order to deliver a high degree of interoperability between a broad range of applications that require such a representation. CWMNX is designed to leverage as much as possible of the existing MusicXML schema.

We currently call this MNX-Common. I don't think anyone would argue with the above definition.

>The GMNX format (working title), designed to represent instantiations of scores in terms of specific visual, performance and audio content and a set of relationships between these facets. GMNX aims to supply a high degree of interoperability to applications that do not depend on particular notational systems or their semantics.

This is what the current spec calls MNX-Generic. I don't think there's anyone here still interested in discussing a non-semantic MNX-Generic format (designed for "applications that do not depend on particular notational systems or their semantics"). There are even good reasons why its very existence would be problematic, and even the co-chair seems to want to push it onto the back burner. Since there's nobody here who wants to discuss it, I think it should simply be dropped. Otherwise it will just fade away for lack of interest.
However, I think MNX-Common should be designed with its output format in mind, so that it becomes easy to implement. The CG's third deliverable should therefore be a generic output format. I'll have a think about how it might be described in the charter: The definition should obviously say that it is openly extensible, and that it uses profiles, one of which is served by MNX-Common.

I would like to propose that we put the charter on the agenda for the coming face-to-face meeting in Frankfurt. It obviously needs updating.

**Comment by mgood** [comment]
@notator you mean @mdgood not me

**Comment by notator** [comment]
Ooops sorry! I won't do it again!

**Comment by notator** [comment]
@mdgood:
>This issue is about splitting MNX-Common from MNX-Generic. If we do this, the MNX container format will likely evolve from its original design, but it will not become redundant or obsolete.

Okay. So this issue affects the MNX container format as well.
I've been looking at the [MNX-Container definition in the Current Draft Spec](https://w3c.github.io/mnx/specification/#mnx-container), and think it would be helpful if we could dscuss two things in particular:
1. The difference between what the Spec calls a [_notational idiom_](https://w3c.github.io/mnx/specification/#notational-idiom) and what it calls a [_profile_](https://w3c.github.io/mnx/specification/#score%20profiles). I don't think we need both concepts.
The criterion for having a particular _profile_ (aka _notational idiom_) would be that it would be provide a meaningful framework within which an application can be written for users that don't need (all) the features defined in some other profile. Those applications would then be easier to write, and the number of editors that can cope with a particular profile increases. An editor that is designed for Church music should not have to be able to cope with Ferneyhough. On the other hand, an editor that can cope with Ferneyhough can probably cope with a whole range of other profiles -- including non-CWMN profiles such as piano-roll notation. It would be up to each editor's vendors to document which mnx music notation profiles their editor can import and/or export. After all, it is the vendors who decide who their target consumers are.
Profiles could very well be defined in terms of other profiles, but I see no advantage to having profiles nested inside _notational idioms_ in the mnx file structure.
 
2. The definition of CWMN
We've tried unsuccessfully to define CWMN in the past, but I think it could be done simply by first deciding on a basic profile (i.e. the minimum that a notation editor needs to support for it to qualify as being a CWMN editor), then defining more (or less) comprehensive profiles as is thought necessary.
I don't think we need to worry about deciding all possible (CWMN) profiles at once, since profiles can always be added later without affecting the existing ones.

An advantage of not nesting profiles in MNX files is that it makes it easier to provide an informative, extensible, file-naming convention.
There could, for example, be defined prefixes to the suffixes like this:

Uninstantiated mnx files:
"Winterreise-cw1.mnx " (=basic CWMN)
"PierrotLunaire-cw.mnx " (= full, normal CWMN)
"FerneyhoughStringQuartet-cw2.mnx " (=advanced CWMN)
"NancarrowStudyVII-pr.mnx " (=piano-roll)
"StockhausenStudy2-gr.mnx " (=a graphical type)
 
The corresponding, SVG instantiations would then be:
"Winterreise-cw1.svg" (=basic CWMN)
"PierrotLunaire-cw.svg" (= full, normal CWMN)
"FerneyhoughStringQuartet-cw2.svg" (=advanced CWMN)
"NancarrowStudyVII-pr.svg" (=piano-roll)
"StockhausenStudy2-gr.svg" (=a graphical type)

Note that off-line applications can also parse SVG! This is not _just_ for the web.

**Comment by adrianholovaty** [comment]
Thanks, all, for the thoughts. Looks like we have general agreement that the specs should be separate, so we'll go ahead and do that. The naming decision (i.e., finding a permanent name for the codenames MNX-Common and MNX-Generic) is a separate issue for another time.

We'll leave this issue open until the work of splitting the specs has been completed.

**Comment by adrianholovaty** [comment]
A quick update: I've run into a problem with our automated tooling (travis-ci and bikeshed), which is blocking the creation of a new spec document. Currently in the process of getting that fixed, at which point the spec will be properly split.

**Comment by shoogle** [comment]
@adrianholovaty, I'd just create a new repo for the Generic spec and keep it entirely separate to the Common spec. That makes the Git History much easier to read and prevents issues and PRs getting mixed up. (Of course you can still refer to Generic issues from Common issues, and vice versa.)

If you want you could the Common and Generic repos as submodules within an overaching MNX repo, but I don't think there is any real need for this.

**Comment by adrianholovaty** [comment]
@shoogle Great point about the revision history — I'll do that! Thanks. :)

**Comment by shoogle** [comment]
@adrianholovaty, glad to help!

No doubt Issue No. 1 for the new repo will be *"What shall we call this repo?"* 😉

**Comment by adrianholovaty** [comment]
Just a note for people following this issue: I've done the splitting work and opened pull request #157. Please give feedback there!

**Comment by adrianholovaty** [comment]
Closing this as done, now that #157 is merged.
